Makefile

```diff
@@ -2134,6 +2124,22 @@ endif
 
 endif
 
+ifeq ($(ARCH), i386)
+EXTRA_CFLAGS += -mhard-float
+EXTRA_CFLAGS += -DMARK_KERNEL_PFU
+else ifeq ($(ARCH), x86_64)
+EXTRA_CFLAGS += -mhard-float
+EXTRA_CFLAGS += -DMARK_KERNEL_PFU
+else ifeq ($(ARCH), arm)
+# Raspbian kernel is with soft-float.
+# 'softfp' allows FP instructions, but no FP on function call interfaces
+ifeq ($(CONFIG_PLATFORM_ARM_RPI), y)
+EXTRA_CFLAGS += -mfloat-abi=softfp
+else
+EXTRA_CFLAGS += -mfloat-abi=hard
```

Comment:
Should we really enforce this on `arm` when `CONFIG_PLATFORM_ARM_RPI` is `n`? Or does it actually work only `arm64`?

Somehow it gives the following error:
https://github.com/aircrack-ng/rtl8812au/issues/802#issuecomment-754014418

Strangely, ArchLinuxARM seems to have no problem building its packages with `-march=armv7-a -mfloat-abi=hard -mfpu=vfpv3-d16` on `armv7h`.